src/xunit.runner.visualstudio/Utility/TestCaseFilter.cs

```diff
@@ -44,7 +56,7 @@ public bool MatchTestCase(TestCase testCase)
         public object PropertyProvider(TestCase testCase, string name)
         {
             // Traits filtering
-            if (knownTraits.Contains(name))
+            if (isDiscovery || knownTraits.Contains(name))
```

Comment:
@bradwilson I probably did not explain myself well with the trait filter. What I meant is that in the line above the filter checks for known traits. If this hashset is empty it will not consider any traits. I can either skip the Contains check as I do now, and make the filtering a bit slower, OR add to the known traits all traits from the current batch, before processing that batch. 

The latter solution seems to be optimal, but I would rather add to the known traits via method, than by passing a live reference to a hashset, because that seems more obvious and less fragile. Do you agree?

Reply:
I don't obsess too much with optimization like this unless it's demonstrably poor. I usually rely on the .NET or Roslyn team to come back and tell me where they've hit performance issues. 😂